### ISSUE TITLE
Cannot upload file with fileName that contains extra dots

### ISSUE THREAD

**Issue by vanhumbeecka** [googleapis/nodejs-storage#1262]
#### Environment details

  - OS: MacOS 0.15.6 (19G73)
  - Node.js version: 10.16.3
  - npm version: 6.9.0
  - `@google-cloud/storage` version: 5.1.2

#### Steps to reproduce

After a frustrating long search, where the error messages pointed me in the wrong direction, I finally managed pinpoint an annoying bug. Furthermore, I discover some other potential bugs, which I will describe below.

Considering the following script:

```
const path = require('path')
const gcs = require('@google-cloud/storage')

class Publisher {

  constructor(jsonFileName) {
    this.storage = new gcs.Storage({
      autoRetry: true,
      keyFilename: jsonFileName,
    });
    this.bucket = this.storage.bucket("releases.my.app");
  }

  async upload(filepath) {
    console.log('electron-publisher-custom', filepath);
    const fileName = path.basename(filepath)
    // // e.g. latest-mac.yml should not be cached for long
    const shortCache = fileName.endsWith(".yml")

    try {
      await this.bucket.upload(filepath, {
        // Support for HTTP requests made with `Accept-Encoding: gzip`
        gzip: true,
        // By setting the option `destination`, you can change the name of the
        // object you are uploading to a bucket.
        destination: fileName,
        predefinedAcl: undefined,
        metadata: {
          // Enable long-lived HTTP caching headers
          // Use only if the contents of the file will never change
          // (If the contents will change, use cacheControl: 'no-cache')
          cacheControl: shortCache ? 'public, max-age=3600' : 'public, max-age=31536000'
        },
      });
    } catch (e) {
      console.error(e)
    }
  }

};

const main = async () => {
  const pub = new Publisher("/path/to/my.json")
  try {
    await pub.upload("/path/to/my/file/to/upload.png")
  } catch (e) {
    console.error(e)
  }
}

main()
```

This script works well in certain cases. But in some cases, it throws a 400 error with message: `Cannot insert legacy ACL for an object when uniform bucket-level access is enabled.` The problem however has nothing to do with the suggested legacy ACL (i've tried every configuration of my bucket, without succes).

The problem is located in the `destination: fileName` in the script above. __if the `fileName` contains extra dots, this error is triggered. Choosing a name without extra dots resolves the problem.__
Example: a `fileName` of `mypicture.png` is no problem
Example2: a `fileName` of `My App-0.0.6.AppImage` returns a 400 error with the legacy ACL message.

This is pretty annoying since these kinds of filenames are the default once that `electron` apps create during publishing. 

OTHER POTENTIAL BUGS:
- Even though I explicitaly state that `predefinedAcl` should be `undefinied`, the library promptly adds `publicRead` in his request. I'm clueless why this is the case.

**Comment by vanhumbeecka** [comment]
Interestingly enough, changing the `destintation` parameter to `encodeURIComponent(fileName)` works, but this has the downside that spaces in the filename are wrong when uploaded.

Example: `encodeURIComponent("My App-0.0.6.AppImage")` equals `My%20App-0.0.6.AppImage` and is named on GCS as such

**Comment by JustinBeckwith** [comment]
Greetings @vanhumbeecka!  I am having a little trouble reproducing this one.  I'm trying to boil it down to the smallest possible reproduction code:

```js
const {Storage} = require('@google-cloud/storage');
const storage = new Storage();
async function main() {
  const bucketName = 'artifacts.el-gato.appspot.com';
  const fileName = 'My App-0.0.6.AppImage'
  const bucket = storage.bucket(bucketName);
  const [, metadata] = await bucket.upload(fileName, {
    fileName
  });
  console.log(metadata);
}
main();
```

Using version `5.1.2` of `@google-cloud/storage`, the code I pasted above seems to work 🤔 Can you spot anything interesting you think may be different?

**Comment by vanhumbeecka** [comment]
Hi there, thanks for looking into it.
Back then I got around the issue, by changing some of my configuration that no spaces were introduced in the filename so I haven't looked at this issue since.

I've now tried to recreate this issue with the code you provided, and everything seems to work as it should. By now, also can't reproduce my original issue with to original code provided on top.

So I'm not really sure what happened back then...
Anyway, since I can't reproduce it myself, i'm closing this issue.
